pkg/config/middleware.go

```diff
@@ -0,0 +1,136 @@
+package config
+
+import (
+	"context"
+	"encoding/json"
+	"log"
+
+	// "encoding/json"
+	"fmt"
+
+	// "log"
+	"net/http"
+	"os"
+	"path/filepath"
+
+	authv1 "k8s.io/api/authentication/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/clientcmd"
+	"k8s.io/klog"
+)
+
+var kClientset *kubernetes.Clientset
+
+func KubeClient() string {
+	config, err, token := getClientConfig()
+	if err != nil {
+		klog.Fatal(err.Error())
+	}
+	kClientset, err = kubernetes.NewForConfig(config)
+
+	if err != nil {
+		klog.Fatal(err.Error())
+	}
+	return token
+
+}
+
+func getKubeConfigPath() string {
+	defaultKubePath := filepath.Join(os.Getenv("HOME"), ".kube", "config")
+	if _, err := os.Stat(defaultKubePath); os.IsNotExist(err) {
+		// set default to empty string if path does not reslove
+		defaultKubePath = ""
+	}
+
+	kubeConfig := getEnv("KUBECONFIG", defaultKubePath)
+	return kubeConfig
+}
+
+func getClientConfig() (*rest.Config, error, string) {
+	kubeConfigPath := getKubeConfigPath()
+	var clientConfig *rest.Config
+	var clientConfigError error
+
+	if kubeConfigPath != "" {
+		klog.Infof("Creating k8s client using KubeConfig at: %s", kubeConfigPath)
+		clientConfig, clientConfigError = clientcmd.BuildConfigFromFlags("", kubeConfigPath)
+	} else {
+		klog.V(2).Info("Creating k8s client using InClusterClientConfig")
+		clientConfig, clientConfigError = rest.InClusterConfig()
+	}
+
+	if clientConfigError != nil {
+		klog.Fatal("Error getting Kube Config: ", clientConfigError)
+	}
+
+	token := clientConfig.BearerToken
+	fmt.Printf("The bearer token is: %s\n", token)
+
+	return clientConfig, clientConfigError, token
+}
+
+//verifies token (userid) with the TokenReview:
+func Middleware() func(http.Handler) http.Handler {
+	return func(next http.Handler) http.Handler {
+		return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+
+			token := KubeClient()
+			fmt.Printf("The token is: %s", token)
```

Comment:
This is our **Search API's Token**.  However, here we are in the context of a user request.  The user's token should be in the Authorization Header or cookie.
```go
r.Header.Get("Authorization")
```

Reply:
Here's the doc to read the cookies (I believe that requests from the UI have it in a cookie instead of the header.
- https://pkg.go.dev/net/http#Request.Cookie

Reply:
Okay, wasn't sure here because I didn't see authorization header in request but did see cookie, so we can use that.

Reply:
We'll need to check both to cover the different scenarios.